pyracf/common/logger.py

```diff
@@ -93,33 +95,65 @@ def log_debug(self, header_message: str, message: str) -> None:
         )
         print(f"{header}\n{message}")
 
-    def redact_strings(
+    def __redact_request_dictionary(
         self,
-        string: Union[str, bytes],
-        redact_strings: List[str],
-    ) -> str:
-        """Redact a list of strings or sequences of bytes in a string or bytes object"""
-        if not redact_strings:
-            return string
-        for to_redact in redact_strings:
-            if isinstance(string, bytes):
-                string = self.redact_string(
-                    string, redact_string=bytes(to_redact, "utf-8")
-                )
-            else:
-                string = self.redact_string(string, redact_string=to_redact)
-        return string
+        dictionary: dict,
+        secret_traits: dict,
+    ) -> dict:
+        """Redact a list of specified secret traits in a request dictionary"""
+        for seg_trait in secret_traits:
+            segment = seg_trait.split(":")[0]
+            if not dictionary.get(segment, {}).get(seg_trait, {}).get("value", None):
+                continue
+            dictionary[segment][seg_trait]["value"] = "********"
+        return dictionary
+
+    def __redact_string(
+        self,
+        input_string: Union[str, bytes],
+        start_ind: int,
+        end_pattern: str,
+    ):
+        """Redacts characters in a string between a starting index and ending pattern"""
+        """Replaces the identified characters with ******** (regardless of length)"""
+        pre_keyword = input_string[:start_ind]
+        post_keyword = end_pattern.join(input_string[start_ind:].split(end_pattern)[1:])
+        return pre_keyword + "********" + end_pattern + post_keyword
 
-    def redact_string(
-        self, string: Union[str, bytes], redact_string: Union[str, bytes]
+    def redact_request_xml(
+        self,
+        xml_string: Union[str, bytes],
+        secret_traits: dict,
+    ) -> Union[str, bytes]:
+        """Redact a list of specific secret traits in a request xml string or bytes object"""
+        """Bases this on xml pattern <xml_key any-operation>xml_value</xml_key>"""
```

Comment:
```python
        """
        Redact a list of specific secret traits in a request xml string or bytes object.
        Based the following xml pattern:
            '<xmltag attribute="any">xml value</xmltag>'
        """
```

Reply:
That is a cleaner way of this level of docstring. I will make similar adjustments elsewhere.

Reply:
Based on below comment on whitespace, I changed this to.

```python
        """
        Redact a list of specific secret traits in a request xml string or bytes object.
        Based the following xml pattern:
            '<xmltag attribute="any">xml value</xmltag>'
        This function also accounts for varied amounts of variation in the xml attributes.
        """
```

Reply:
My guess is that the last comment is saying that it accounts for any number of arbitrary xml attributes?

```python
        """
        Redact a list of specific secret traits in a request xml string or bytes object.
        Based the following xml pattern:
            '<xmltag attribute="any">xml value</xmltag>'
        This function also accounts for any number of arbitrary xml attributes.
        """
```

Reply:
I didn't realize I put varied and variation in the comment. I wanted to leave it more general because even no attributes or improperly defined ones will be ignored. it really looks for <xmltag[ANY or NO TEXT HERE]>. I think what you wrote is more than clear enough, so I will fix that.

Reply:
I am seeing the changes now.